src/util.c

```diff
@@ -149,6 +149,26 @@ int write_zeroes(int fd, size_t l) {
         return 0; /* Return == 0 if we could only write out zeroes */
 }
 
+static bool memeqzero(const void *data, size_t length) {
+        /* Does the buffer consist entirely of NULs?
+         * Copied from https://github.com/rustyrussell/ccan/blob/master/ccan/mem/mem.c#L92.
+         */
+
+        const unsigned char *p = data;
```

Comment:
so far we used uint8_t as generic type meaning "any byte", if figure we should use that here too, even if this os copied from ccan...

btw, what about the license there? if figure we need to at least mention where this comes from and who wrote it...

Reply:
I'll switch to uint8_t.

Upstream license is CC-0, so it's OK to copy, but I should mention that in the commit message.